vacation/src/main/java/cz/fi/muni/pv217/employee/management/VacationResource.java

```diff
@@ -0,0 +1,157 @@
+                    .entity(String.format("Vacation with id %d not found.", id))
+                    .build();
+        }
+        return Response.ok(vacation).build();
+    }
+
+    @GET
+    @Path("{id}")
+    @APIResponses(
+        value = {
+            @APIResponse(
+                responseCode = "404",
+                description = "Vacation does not exists"),
+            @APIResponse(
+                responseCode = "401",
+                description = "You don't have permissions to access said vacation"),
+            @APIResponse(
+                responseCode = "200",
+                description = "Particular vacation.",
+                content = @Content(mediaType = "application/json",
+                schema = @Schema(implementation = Vacation.class)))})
+    @Counted(absolute = true, name = "vacation.get")
+    public Vacation get(@PathParam("id") long id) {
+        Optional<Vacation> optionalVacation = Vacation.findByIdOptional(id);
+        Vacation vacation = optionalVacation.orElseThrow(() ->
+                new WebApplicationException("Vacation with id " + id + " does not exist.", 404));
+        Employee employee = coreService.getEmployee(vacation.employee_id);
+        if (!identity.hasRole("admin") && !identity.getPrincipal().getName().equals(employee.username)) {
+            throw new WebApplicationException("You don't have permission to access this resource", 403);
+        }
+        return vacation;
+    }
+
+    @GET
+    @RolesAllowed("admin")
+    @Counted(absolute = true, name = "vacation.getAll")
+    public List<Vacation> getAll(@Context SecurityContext sec) {
+        return Vacation.listAll();
+    }
+
+    @GET
+    @Path("employee/{id}")
+    @Counted(absolute = true, name = "vacation.getByEmployee")
+    public List<Vacation> getByEmployee(@PathParam("id") long id,
+                                        @QueryParam("from") Optional<String> optionalFrom,
+                                        @QueryParam("to") Optional<String> optionalTo) {
+        LocalDate from = optionalFrom.map(LocalDate::parse).orElse(LocalDate.now().withDayOfMonth(1));
+        LocalDate to = optionalTo.map(LocalDate::parse).orElse(LocalDate.now());
+
+        List<Vacation> vacations = Vacation.list("employee_id", id);
+        List<Vacation> result = new LinkedList<>();
+
+        for (Vacation elem : vacations) {
+            if (elem.date.isAfter(from) && elem.date.isBefore(to)) {
+                result.add(elem);
+            }
+        }
+        return result;
+    }
+
+    @GET
+    @Path("employee/{id}/date")
+    @Counted(absolute = true, name = "vacation.hasEmployeeVacationOnDate")
+    public boolean hasEmployeeVacationOnDate(@PathParam("id") long id,
+                                             @QueryParam("date") String date) {
+        LocalDate targetDate = LocalDate.parse(date);
+        List<Vacation> vacations = Vacation.list("employee_id", id);
+
+        for (Vacation elem : vacations) {
```

Comment:
You coul use something like this `Vacation.find("date = :date", Parameters.with("date", targetDate))` https://quarkus.io/guides/hibernate-orm-panache#query-parameters.